### PR TITLE
Fixed problem with path functions not keeping prefix (MLDB-1781)

### PR DESCRIPTION
**Pull request by jeremybarnes**

### CODE REVIEW

**Comment by jeremybarnes** [comment]
Test failures are unrelated to this change.

File: sql/expression_value.cc [hunk 1]
```diff
@@ -1780,13 +1780,17 @@ ExpressionValue(RowValue row) noexcept
                        RowValue::iterator last,
                        size_t level)
         {
+            cerr << "doing level " << level << " from "
```
Comment (mailletf) [inline sql/expression_value.cc]: comment

Comment (jeremybarnes) [inline sql/expression_value.cc]: Right... removed.

**Comment by mailletf** [comment]
+1 based on the test

**Pull request merged**